python/pyarrow/table.pxi

```diff
@@ -160,6 +160,8 @@ cdef class ChunkedArray(_PandasConvertible):
             return _normalize_slice(self, key)
         elif isinstance(key, int):
```

Comment:
I think we could use the python conversion helper  [CIntFromPython](https://github.com/apache/arrow/blob/master/cpp/src/arrow/python/helpers.h#L92) which should handle numpy scalars as well.